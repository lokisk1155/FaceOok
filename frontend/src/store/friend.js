import csrfFetch from "./csrf";
export const REMOVE_FRIEND = "friends/REMOVE_FRIEND";
export const RECEIVE_FRIEND = "friends/RECEIVE_FRIEND";
export const RECEIVE_FRIENDS = "friends/RECEIVE_FRIENDS";

export const removeFriend = (friendship) => ({
  type: REMOVE_FRIEND,
  friendship,
});
export const receiveFriend = (payload) => ({
  type: RECEIVE_FRIEND,
  payload,
});

export const receiveFriends = (friends) => ({
  type: RECEIVE_FRIENDS,
  payload: friends,
});

export const fetchFriend = (userId) => async (dispatch) => {
  const res = await csrfFetch(`/api/friends?userId=${userId}`);
  const data = await res.json();
  if (res.ok) {
    dispatch(receiveFriend({ friend: { [data.id]: data } }));
  }
};

export const fetchFriends = (userId) => async (dispatch) => {
  const res = await csrfFetch(`/api/friends?userId=${userId}`);
  if (res.ok) {
    const fetchedUsers = await res.json();
    dispatch(receiveFriends(fetchedUsers));
  }
};

export const addFriend = (friendRequest) => async (dispatch) => {
  const res = await csrfFetch(`/api/friends`, {
    method: "POST",
    body: JSON.stringify(friendRequest),
  });
  const data = await res.json();
  dispatch(receiveFriend(data));
};

export const deleteFriend = (userId) => async (dispatch) => {
  const res = await csrfFetch(`/api/friends/${userId}`, {
    method: "DELETE",
  });
  const data = await res.json();
  dispatch(removeFriend(data));
};

const friendReducer = (previousState = {}, action) => {
  let newState = {};
  switch (action.type) {
    case RECEIVE_FRIEND:
      newState = { ...previousState, ...action.payload.friend };
      return newState;
    case REMOVE_FRIEND:
      newState = { ...previousState };
      delete newState[action.friendship.id];
      return newState;
    case RECEIVE_FRIENDS:
      newState = { ...action.payload };
      return newState;
    default:
      return previousState;
  }
};

export default friendReducer;
