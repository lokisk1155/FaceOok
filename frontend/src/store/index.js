import { createStore, combineReducers, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import friendReducer from './friend';
import postsReducer from './post';
import sessionReducer from './session';
import userReducer from './user';

export const rootReducer = combineReducers({
    session: sessionReducer,
    user: userReducer,
    post: postsReducer,
    friend: friendReducer
})

let enhancer;

if (process.env.NODE_ENV === 'production') {
  enhancer = applyMiddleware(thunk);
} else {
  const logger = require('redux-logger').default;
  const composeEnhancers =
    window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
  enhancer = composeEnhancers(applyMiddleware(thunk, logger));
}

const configureStore = (preloadedState = {}) => {
    return createStore(rootReducer, preloadedState, enhancer);
};

export default configureStore