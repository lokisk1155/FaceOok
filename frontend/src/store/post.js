import csrfFetch from "./csrf";

const RECEIVE_POST = "post/receivePost";
const RECEIVE_POSTS = "post/receivePosts";
const REMOVE_POST = "post/removePost";

export const receivePost = (post) => ({
  type: RECEIVE_POST,
  payload: post,
});

export const receivePosts = (posts) => ({
  type: RECEIVE_POSTS,
  payload: posts,
});

export const removePost = (postId) => ({
  type: REMOVE_POST,
  payload: postId,
});

export const fetchPost = (postId) => async (dispatch) => {
  const postRes = await csrfFetch(`/api/posts/${postId}`);
  const postData = await postRes.json();
  return postData;
};

export const fetchPosts = (limit) => async (dispatch) => {
  let postRes;
  if (limit) {
    postRes = await csrfFetch(`/api/posts?limit=${limit}`);
  } else {
    postRes = await csrfFetch(`/api/posts`);
  }
  const postData = await postRes.json();
  dispatch(receivePosts(postData));
};

export const createPost = (post) => async (dispatch) => {
  const postRes = await csrfFetch("/api/posts", {
    method: "POST",
    body: JSON.stringify(post),
  });
  const postData = await postRes.json();
  dispatch(receivePosts(postData));
  return postData;
};

export const updatePost = (post) => async (dispatch) => {
  const postRes = await csrfFetch(`/api/posts/${post.id}`, {
    method: "PUT",
    body: JSON.stringify({ post }),
  });
  const postData = await postRes.json();
  dispatch(receivePost(postData));
  return postData;
};

export const deletePost = (postId) => async (dispatch) => {
  await csrfFetch(`/api/posts/${postId}`, { method: "DELETE" });
  dispatch(removePost(postId));
};

const postsReducer = (previousState = {}, action) => {
  let newState = { ...previousState };
  switch (action.type) {
    case RECEIVE_POST:
      return { ...newState, [action.payload.id]: action.payload };
    case RECEIVE_POSTS:
      newState = { ...action.payload };
      return newState;
    case REMOVE_POST:
      delete newState[action.payload.postId];
      return newState;
    default:
      return previousState;
  }
};

export default postsReducer;
