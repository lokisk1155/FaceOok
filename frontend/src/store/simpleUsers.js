import csrfFetch from "./csrf";

export const SET_SIMPLE_USERS = "users/SET_SIMPLE_USERS";

export const setSimpleUsers = (users) => ({
  type: SET_SIMPLE_USERS,
  payload: users,
});

export const getUsers = () => async (dispatch) => {
  const res = await csrfFetch(`/api/users`);
  const data = await res.json();
  const filtered = Object.values(data).map((user) => {
    return {
      id: user.id,
      name: `${user.first_name} ${user.last_name}`,
      photo: user.photo,
    };
  });

  dispatch(setSimpleUsers(filtered));
  return data;
};

const simpleUsersReducer = (previousState = {}, action) => {
  let newState = { ...previousState };
  switch (action.type) {
    case SET_SIMPLE_USERS:
      newState = { ...action.payload };
      return newState;
    default:
      return previousState;
  }
};

export default simpleUsersReducer;
