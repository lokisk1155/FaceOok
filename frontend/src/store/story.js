import csrfFetch from "./csrf";

const ADD_STORY = "stories/ADD_STORY";

export const addStory = (story) => ({
  type: ADD_STORY,
  payload: story,
});

export const createStory = (userId) => async (dispatch) => {
  const storyRequest = await csrfFetch(`/api/stories?userId=${userId}`, {
    method: "POST",
  });
  const storyResponse = await storyRequest.json();
  return dispatch(addStory(storyResponse));
};

export const storiesReducer = (previousState = {}, action) => {
  switch (action.type) {
    case ADD_STORY:
      return { ...previousState, [action.payload.id]: action.payload };
    default:
      return previousState;
  }
};
