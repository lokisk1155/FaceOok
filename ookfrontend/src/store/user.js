import csrfFetch from './csrf';

export const SET_CURRENT_PROFILE = "users/SET_CURRENT_PROFILE"

export const setCurrentProfile = (user) => ({
      type: SET_CURRENT_PROFILE,
      user
})

export const fetchUser = userId => async dispatch => {
    const res = await csrfFetch(`/api/users/${userId}`)
  
      const user = await res.json() 
      dispatch(setCurrentProfile(user))
}


const userReducer = (previousState = {}, action) => {
    let newState = {...previousState}
      switch(action.type) {
        case SET_CURRENT_PROFILE: 
            newState = action.user
            return newState 
        default:
            return previousState
    }
}

export default userReducer