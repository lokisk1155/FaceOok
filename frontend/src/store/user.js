import csrfFetch from "./csrf";
import { RECEIVE_FRIEND, REMOVE_FRIEND } from "./friend";

export const SET_CURRENT_PROFILE = "users/SET_CURRENT_PROFILE";
export const RECEIVE_USERS = "users/RECEIVE_USERS";
export const SET_CURRENT_ARRAY = "users/SET_CURRENT_ARRAY";

export const setCurrentProfile = (user) => {
  return {
    type: SET_CURRENT_PROFILE,
    payload: user,
  };
};

export const setCurrentArray = (user) => {
  return {
    type: SET_CURRENT_ARRAY,
    payload: user,
  };
};

export const receiveUsers = (users) => ({
  type: RECEIVE_USERS,
  payload: users,
});

export const fetchUsers = () => async (dispatch) => {
  const res = await csrfFetch(`api/users`);
  const data = await res.json();
  dispatch(receiveUsers(data));
};

export const fetchUser = (userId) => async (dispatch) => {
  const res = await csrfFetch(`/api/users/${userId}`);

  const data = await res.json();
  dispatch(setCurrentProfile(data.user));
};

export const getCurrent = (id) => (state) => {
  return state.user[id];
};

export const updateUser = (user) => async (dispatch) => {
  const res = await csrfFetch(`/api/users/${user.id}`, {
    method: "PUT",
    body: JSON.stringify({ user }),
  });
  const data = await res.json();
  dispatch(setCurrentProfile(data.user));
  return data;
};

export const updateUserArray = (user) => async (dispatch) => {
  const res = await csrfFetch(`/api/users/${user.id}`, {
    method: "PUT",
    body: JSON.stringify({ user }),
  });
  const data = await res.json();
  dispatch(setCurrentArray(data.user));
  return data;
};

const userReducer = (previousState = {}, action) => {
  let newState = { ...previousState };
  switch (action.type) {
    case SET_CURRENT_PROFILE:
      newState = { ...previousState, [action.payload.id]: action.payload };
      return newState;
    case SET_CURRENT_ARRAY:
      newState = { ...previousState, [action.payload.id]: action.payload };
      return newState;
    case RECEIVE_USERS:
      return { ...previousState, ...action.payload };
    case RECEIVE_FRIEND:
      return { ...previousState, ...action.payload.user };
    // case REMOVE_FRIEND:
    //   debugger;

    default:
      return previousState;
  }
};

export default userReducer;
