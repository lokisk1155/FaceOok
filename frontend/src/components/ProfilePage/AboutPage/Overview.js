import { updateUser } from "../../../store/user"
import { useDispatch } from "react-redux"
import { useParams } from "react-router-dom"
import { getCurrent } from "../../../store/user"
import { useSelector } from "react-redux"
import { fetchUser } from "../../../store/user"
import { useEffect, useState } from "react"


function Overview({ currentUser, isUser }) {
    const dispatch = useDispatch()

    const [workPlace, setWorkPlace] = useState('')
    const [toggleWork, setToggleWork] = useState(false)
    const [fakeWork, setFakeWork] = useState('')
    const [toggleWorkEdit, setToggleWorkEdit] = useState(false)

    const [education, setEducation] = useState('')
    const [toggleEducation, setToggleEducation] = useState(false)
    const [fakeEducation, setFakeEducation] = useState('')
    const [toggleEducationEdit, setToggleEducationEdit] = useState(false)

    const [location, setLocation] = useState('')
    const [toggleLocation, setToggleLocation] = useState(false)
    const [fakeLocation, setFakeLocation] = useState('')
    const [toggleLocationEdit, setToggleLocationEdit] = useState(false)

    const [relationship, setRelationShip] = useState('')
    const [toggleRelationship, setToggleRelationship] = useState(false)
    const [fakeRelationship, setFakeRelationship] = useState('')
    const [toggleRelationshipEdit, setToggleRelationshipEdit] = useState(false)


    useEffect(() => {
        checkParams()
    }, [])

    function checkParams() {
        if (currentUser) {
            setWorkPlace(currentUser.work)
            setRelationShip(currentUser.relationship)
            setEducation(currentUser.education)
            setLocation(currentUser.location)
        }
    }


    const handleWork = () => {
        setWorkPlace(fakeWork)
        let work = fakeWork
        const user = {
            ...currentUser, work
        }
        return dispatch(updateUser(user))
    }

    const handleEducation = () => {
        setEducation(fakeEducation)
        let education = fakeEducation
        const user = {
            ...currentUser, education
        }
        return dispatch(updateUser(user))
    }

    const handleLocation = () => {
        setLocation(fakeLocation)
        let location = fakeLocation 
        const user = {
            ...currentUser, location
        }
        return dispatch(updateUser(user))
    }

    const handleRelationship = () => {
        setRelationShip(fakeRelationship)
        let relationship = fakeRelationship
        const user = {
            ...currentUser, relationship
        }
        return dispatch(updateUser(user))
    }

    function handleClick() {
        return setToggleWorkEdit(true)
    }


    return (
        <div>
            <div>
                <p>{workPlace}</p>
                {!workPlace && isUser && <button onClick={(() => {
                    setToggleWork(true)
                })}>Add a workplace</button>}
                {toggleWork && !workPlace &&
                <form onSubmit={handleWork}>
                    <input type="text" placeholder="Company" onChange={((e) => setFakeWork(e.target.value))}/>
                    <button onClick={(() => setToggleWork(false))}>cancel</button>
                    <input type="submit" value="save"/>
                </form>}

                {isUser && workPlace && <button onClick={handleClick}>Edit Post</button>}
                {toggleWorkEdit && <form onSubmit={handleWork}>
                    <input type="text" placeholder="Company" onChange={((e) => setFakeWork(e.target.value))}></input>
                    <button onClick={(() => setToggleWorkEdit(false))}>cancel</button>
                    <input type="submit" value="save"/>
                </form>}
            </div>

            <div>
                <p>{education}</p>
                {!education && isUser && <button onClick={(() => {
                    setToggleEducation(true)
                })}>Add Education</button>}
                {toggleEducation && !education && 
                 <form onSubmit={handleEducation}>
                    <input type="text" placeholder="Institution" onChange={((e) => setFakeEducation(e.target.value))}/>
                    <button onClick={(() => setToggleEducation(false))}>cancel</button>
                    <input type="submit" value="save"/>
                </form>}

                {isUser && education && <button onClick={(() => setToggleEducationEdit(true))}>Edit Education</button>}
                {toggleEducationEdit && <form onSubmit={handleEducation}>
                    <input type="text" placeholder="Institution" onChange={((e) => setFakeEducation(e.target.value))}></input>
                    <button onClick={(() => setToggleEducationEdit(false))}>cancel</button>
                    <input type="submit" value="save"/>
                </form>}
            </div>

            <div>
                <p>{location}</p>
                {!location && isUser && <button onClick={(() => {
                        setToggleLocation(true)
                    })}>Add Location</button>}
                    {toggleLocation && !location&& 
                    <form onSubmit={handleLocation}>
                        <input type="text" placeholder="Location" onChange={((e) => setFakeLocation(e.target.value))}/>
                        <button onClick={(() => setToggleLocation(false))}>cancel</button>
                        <input type="submit" value="save"/>
                    </form>}

                    {isUser && location && <button onClick={(() => setToggleLocationEdit(true))}>Change Primary Location</button>}
                    {toggleLocationEdit && <form onSubmit={handleLocation}>
                        <input type="text" placeholder="Location" onChange={((e) => setFakeLocation(e.target.value))}></input>
                        <button onClick={(() => setToggleLocationEdit(false))}>cancel</button>
                        <input type="submit" value="save"/>
                    </form>}
                </div>

            <div>
                <p>{relationship}</p>
            {!relationship && isUser && <button onClick={(() => {
                    setToggleRelationship(!toggleRelationship)
                })}>Add Relationship Status</button>}
                {toggleRelationship && !relationship &&
                <form onSubmit={handleRelationship}>
                    <select type="text" onChange={((e) => setFakeRelationship(e.target.value))}>
                            <option value="Single">Single</option>
                            <option value="In a relationship">In a relationship</option>
                            <option value="Married">Married</option>
                            <option value="In a civil union">In a civil union</option>
                            <option value="In a domestic partnership">In a domestic partnership</option>
                            <option value="In an open relationship">In an open relationship</option>
                            <option value="It's complicated">It's complicated</option>
                            <option value="Seperated">Seperated</option>
                            <option value="Divorced">Divorced</option>
                            <option value="Widowed">Widowed</option>
                    </select>
                    <input type="submit" />
                </form>}

                {isUser && relationship && <button onClick={(() => setToggleRelationshipEdit(true))}>Edit Relationship Status</button>}
                    {toggleRelationshipEdit && <form onSubmit={handleRelationship}>
                    <select type="text" onChange={((e) => setFakeRelationship(e.target.value))}>
                            <option value="Single">Single</option>
                            <option value="In a relationship">In a relationship</option>
                            <option value="Married">Married</option>
                            <option value="In a civil union">In a civil union</option>
                            <option value="In a domestic partnership">In a domestic partnership</option>
                            <option value="In an open relationship">In an open relationship</option>
                            <option value="It's complicated">It's complicated</option>
                            <option value="Seperated">Seperated</option>
                            <option value="Divorced">Divorced</option>
                            <option value="Widowed">Widowed</option>
                    </select>
                    <button onClick={(() => setToggleRelationshipEdit(true))}>cancel</button>
                    <input type="submit" value="save"/>
                </form>}
            </div>

        </div>
    )

}

export default Overview