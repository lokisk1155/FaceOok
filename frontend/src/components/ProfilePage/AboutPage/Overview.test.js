import renderer from 'react-test-renderer';
import { createStore, combineReducers, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import  Overview  from './Overview'
import { Provider } from 'react-redux'
import { Route, MemoryRouter } from 'react-router-dom'
import userReducer from '../../../store/user';
import sessionReducer from '../../../store/session';
import { setCurrentProfile } from '../../../store/user';
import { setCurrentUser } from '../../../store/session';
import TestRenderer from 'react-test-renderer'
import ReactDOMServer from 'react-dom/server';
import { useDispatch } from 'react-redux';

// const dispatch = useDispatch() 

// // dispatch(setCurrentProfile(currentUser))

let enhancer;
const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
enhancer = composeEnhancers(applyMiddleware(thunk));




export const rootReducer = combineReducers({
    session: sessionReducer,
    user: userReducer,
  });


  const configureStore = (preloadedState = {}) => {
    const user = {
            id: 1,
            first_name: 'shawn',
            last_name: 'mallon',
            work: 'dude',
            relationship: 'single',
            education: 'school',
            location: 'katonah'
    }
    const initialState = {
      session: { user: user },
      user: { 1: user} 
    };
    return createStore(rootReducer, { ...preloadedState, ...initialState }, enhancer);
  };
  

const store = configureStore();

it(`renders user data `, async () => {
    const component = ReactDOMServer.renderToString(
        <Provider store={store}>
            <MemoryRouter initialEntries={[`/ProfilePage/1/about`]}>
                <Route path="/ProfilePage/:id/about">
                    <Overview />
                </Route>
            </MemoryRouter>
         </Provider>
    )
    // let tree = component.toJSON();
    //console.log(component, 'root')
    expect(component).toBe('<div class="content-container"><div><p></p><button>Add a workplace</button></div><div><p></p><button>Add Education</button></div><div><p></p><button>Add Location</button></div><div><p></p><button>Add Relationship Status</button></div></div>')
    // .toMatchSnapshot() 

})