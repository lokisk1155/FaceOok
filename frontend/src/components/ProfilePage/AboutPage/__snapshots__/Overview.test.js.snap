// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders user data  1`] = `
<div
  className="content-container"
>
  <div>
    <p />
  </div>
  <div>
    <p />
  </div>
  <div>
    <p />
  </div>
  <div>
    <p />
  </div>
</div>
`;
